src/bridges/indexcoop/IndexBridge.sol

```diff
@@ -0,0 +1,443 @@
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title icETH Bridge
+ * @dev A smart contract responsible for buying and selling icETH with ETH etther through selling/buying from
+ * a DEX or by issuing/redeeming icEth set tokens.
+ */
+contract IndexBridgeContract is BridgeBase {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    error UnsafeOraclePrice();
+
+    address public constant EXCHANGE_ISSUANCE = 0xB7cc88A13586D862B97a677990de14A122b74598;
+    address public constant CURVE = 0xDC24316b9AE028F1497c275EB9192a3Ea0f67022;
+    address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant STETH = 0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84;
+    address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    address public constant ICETH = 0x7C07F7aBe10CE8e33DC6C5aD68FE033085256A84;
+    address public constant STETH_PRICE_FEED = 0xAb55Bf4DfBf469ebfe082b7872557D1F87692Fe6;
+    address public constant AAVE_LEVERAGE_MODULE = 0x251Bd1D42Df1f153D86a5BA2305FaADE4D5f51DC;
+    address public constant CHAINLINK_STETH_ETH = 0x86392dC19c0b719886221c78AB11eb8Cf5c52812;
+
+    address public constant UNIV3_ROUTER = 0xE592427A0AEce92De3Edee1F18E0157C05861564;
+    address public constant UNIV3_FACTORY = 0x1F98431c8aD98523631AE4a59f267346ea31F984;
+
+    uint256 private constant DUST = 1;
+
+    constructor(address _rollupProcessor) BridgeBase(_rollupProcessor) {
+        // Tokens can be pre approved since the bridge should not hold any tokens.
+        IERC20(ICETH).safeIncreaseAllowance(address(EXCHANGE_ISSUANCE), type(uint256).max);
+        IERC20(ICETH).safeIncreaseAllowance(UNIV3_ROUTER, type(uint256).max);
+        IERC20(ICETH).safeIncreaseAllowance(ROLLUP_PROCESSOR, type(uint256).max);
+        IERC20(WETH).safeIncreaseAllowance(UNIV3_ROUTER, type(uint256).max);
+    }
+
+    receive() external payable {}
+
+    /**
+     * @notice Function that swaps between icETH and ETH.
+     * @dev The flow of this functions is based on the type of input and output assets.
+     *
+     * @param _inputAssetA - ETH to buy/issue icETH, icETH to sell/redeem icETH.
+     * @param _outputAssetA - icETH to buy/issue icETH, ETH to sell/redeem icETH.
+     * @param _outputAssetB - ETH to buy/issue icETH, empty to sell/redeem icETH.
+     * @param _totalInputValue - Total amount of ETH/icETH to be swapped for icETH/ETH
+     * @param _interactionNonce - Globablly unique identifier for this bridge call
+     * @param _auxData - Encodes flowSelector and maxSlip. See notes on the decodeAuxData
+     * function for encoding details.
+     * @return outputValueA - Amount of icETH received when buying/issuing, Amount of ETH
+     * received when selling/redeeming.
+     * @return outputValueB - Amount of ETH returned when issuing icETH. A portion of ETH
```

Comment:
Have you tried avoiding this by using getIssueExactSet on the issuance contract? I know they mention there that it is supposed to be used off-chain because of Uni v3 quoter but unless I am mistaken icETH uses curve so there should not be such an issue. Shouldn't be that hard to gas profile it if unsure (forge test --gas-report)